src/main/java/org/example/Main.java

```diff
@@ -121,339 +177,256 @@ public boolean isHappy() {
-                            continue;
-                        }
-                        if (users.contains("JUMPING") && !num.jumping) {
-                            continue;
-                        }
-                        if (users.contains("HAPPY") && !num.happy) {
-                            continue;
-                        }
-                        if (users.contains("SAD") && !num.sad) {
-                            continue;
-                        }
-                        if (users.contains("-EVEN") && num.even) {
-                            continue;
-                        }
-                        if (users.contains("-ODD") && num.odd) {
-                            continue;
-                        }
-                        if (users.contains("-BUZZ") && num.buzz) {
-                            continue;
-                        }
-                        if (users.contains("-DUCK") && num.duck) {
-                            continue;
-                        }
-                        if (users.contains("-PALINDROMIC") && num.palindromic) {
-                            continue;
-                        }
-                        if (users.contains("-SPY") && num.spy) {
-                            continue;
-                        }
-                        if (users.contains("-GAPFUL") && num.gapful) {
-                            continue;
-                        }
-                        if (users.contains("-SUNNY") && num.sunny) {
-                            continue;
-                        }
-                        if (users.contains("-SQUARE") && num.square) {
-                            continue;
-                        }
-                        if (users.contains("-JUMPING") && num.jumping) {
-                            continue;
-                        }
-                        if (users.contains("-HAPPY") && num.happy) {
-                            continue;
-                        }
-                        if (users.contains("-SAD") && num.sad) {
-                            continue;
-                        }
-                        num.printShortInfo();
-                        count--;
-                    }
-                }
+        Set<String> allProperties =
+                Stream.of(MyNumber.Params.values()).map(MyNumber.Params::name).collect(Collectors.toSet());
+        Set<String> allNegativeProperties =
+                Stream.of(MyNumber.Params.values()).map(MyNumber.Params::name).map(x -> "-" + x).collect(Collectors.toSet());
+        Collection<String> incorrectProperties = new ArrayList<>();
+        for (String property : properties) {
+            if (!(allProperties.contains(property) || allNegativeProperties.contains(property))) {
+                incorrectProperties.add(property);
             }
         }
+
+        if (incorrectProperties.isEmpty()) {
+            return false;
+        }
+        if (incorrectProperties.size() == 1) {
```

Comment:
I think just one like is OK:
`System.out.println(MessageFormat.format("The properties [{0}] are wrong.", String.join(", ", incorrectProperties)));`

Reply:
java.lang.AssertionError: Wrong answer in test #29

The request: "1 10 May" has one wrong property. Expected message: "The property ... is wrong".